src/actionmanager.cpp

```diff
@@ -827,6 +835,25 @@ void ActionManager::initializeActionLibrary()
     }
 }
 
+QIcon ActionManager::getCacheableIcon(const QString &cacheKey, const QIcon &icon)
+{
+    static QMutex mutex;
+    static QCache<QString, QIcon> cache;
```

Comment:
Can we not just use the application-wide QPixmapCache for this? I believe that is what the caching on the actions is doing anyway, although I guess we would need to take care to avoid overlapping keys

Reply:
Well this is working on the QIcon level which can contain multiple sizes. Putting them in QPixmapCache would involve an entry for each size (resolution) and then you'd have to somehow know which sizes were placed in there when reconstructing the QIcon to put in the menu. Aside from that, `getOpenWithItems` currently runs in a different thread (via `QtConcurrent::run`), so it wouldn't have access to QPixmapCache. (That's not the source of the performance issues in the first place, by the way, it's just a slow when I tried making it run in the main thread)